goldigen/generator.go

```diff
@@ -206,10 +207,10 @@ func (g *Generator) generateTypeRegistrationFunction(conf *TypesConfiguration, o
 
 func (g *Generator) logVerbose(message string, args ...interface{}) {
 	if g.Debug {
-		fmt.Fprintf(os.Stderr, message+"\n", args...)
+		fmt.Fprintf(osStderr, message+"\n", args...)
 	}
 }
 
 func (g *Generator) logWarn(message string, args ...interface{}) {
-	fmt.Fprintf(os.Stderr, message+"\n", args...)
+	fmt.Fprintf(osStderr, message+"\n", args...)
```

Comment:
I think I would prefer adding a `Logger` field to the `Generator` and `GoPathChecker` instead of introducing a global variable.

```diff
diff --git a/goldigen/generator.go b/goldigen/generator.go
index b4ac9e5..f146b58 100644
--- a/goldigen/generator.go
+++ b/goldigen/generator.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"os"
 	"sort"
 	"strings"
 
@@ -17,6 +18,7 @@ var ioutilReadAll = ioutil.ReadAll
 type Generator struct {
 	Config Config
 	Debug  bool
+	Logger io.Writer
 }
 
 // NewGenerator creates a new Generator instance
@@ -24,6 +26,7 @@ func NewGenerator(config Config) *Generator {
 	return &Generator{
 		Config: config,
 		Debug:  false,
+		Logger: os.Stderr,
 	}
 }
 
@@ -207,10 +210,10 @@ func (g *Generator) generateTypeRegistrationFunction(conf *TypesConfiguration, o
 
 func (g *Generator) logVerbose(message string, args ...interface{}) {
 	if g.Debug {
-		fmt.Fprintf(osStderr, message+"\n", args...)
+		fmt.Fprintf(g.Logger, message+"\n", args...)
 	}
 }
 
 func (g *Generator) logWarn(message string, args ...interface{}) {
-	fmt.Fprintf(osStderr, message+"\n", args...)
+	fmt.Fprintf(g.Logger, message+"\n", args...)
 }
diff --git a/goldigen/generator_test.go b/goldigen/generator_test.go
index 26ce5ca..42dd5cf 100644
--- a/goldigen/generator_test.go
+++ b/goldigen/generator_test.go
@@ -240,12 +240,11 @@ var _ = Describe("Generator", func() {
 	})
 
 	It("should log message in debug mode", func() {
-		stdErrorBytes = []byte(``)
-		osStderr = &stderrMock{}
-
+		logger := new(bytes.Buffer)
 		gen.Debug = true
+		gen.Logger = logger
 		gen.Generate(strings.NewReader(exampleYaml), output)
-		Expect(stdErrorBytes).NotTo(BeEmpty())
+		Expect(logger.String()).NotTo(BeEmpty())
 		gen.Debug = false
 	})
 
diff --git a/goldigen/go_path_checker.go b/goldigen/go_path_checker.go
index 5e81e03..6cfc746 100644
--- a/goldigen/go_path_checker.go
+++ b/goldigen/go_path_checker.go
@@ -3,22 +3,28 @@ package main
 import (
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
 type GoPathChecker struct {
-	Verbose bool
+	Logger io.Writer
 }
 
 var (
 	filepathAbs           = filepath.Abs
-	osStderr    io.Writer = os.Stderr
 )
 
 func NewGoPathChecker(isVerbose bool) *GoPathChecker {
-	return &GoPathChecker{isVerbose}
+	c := &GoPathChecker{Logger: ioutil.Discard}
+
+	if isVerbose {
+		c.Logger = os.Stderr
+	}
+
+	return c
 }
 
 func (c *GoPathChecker) PackageName(outputPath string) string {
@@ -54,7 +60,5 @@ func (c *GoPathChecker) PackageName(outputPath string) string {
 }
 
 func (c *GoPathChecker) log(message string, args ...interface{}) {
-	if c.Verbose {
-		fmt.Fprintf(osStderr, message+"\n", args...)
-	}
+	fmt.Fprintf(c.Logger, message+"\n", args...)
 }
diff --git a/goldigen/go_path_checker_test.go b/goldigen/go_path_checker_test.go
index c342d5e..c1c2fd4 100644
--- a/goldigen/go_path_checker_test.go
+++ b/goldigen/go_path_checker_test.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"bytes"
 	"errors"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
@@ -9,16 +10,6 @@ import (
 	"os"
 )
 
-var stdErrorBytes []byte
-
-type stderrMock struct {
-}
-
-func (s *stderrMock) Write(p []byte) (n int, err error) {
-	stdErrorBytes = p
-	return len(p), nil
-}
-
 var _ = Describe("GoPathChecker", func() {
 	var verbose = false
 
@@ -70,11 +61,11 @@ var _ = Describe("GoPathChecker", func() {
 		})
 
 		It("should log message in verbose mode", func() {
-			stdErrorBytes = []byte(``)
-			osStderr = &stderrMock{}
+			logger := new(bytes.Buffer)
 			checker := NewGoPathChecker(true)
+			checker.Logger = logger
 			checker.PackageName("/home/fgrosse/go/src/foo/bar/baz/goldi.go")
-			Expect(stdErrorBytes).NotTo(BeEmpty())
+			Expect(logger.String()).NotTo(BeEmpty())
 		})
```

Reply:
Done adding Logger.